Fixes for cross-compilation

Artturin opened the pull request:
the docs for bundlerApp used to have makeWrapper in buildInputs but it
has been corrected already

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin committed:
treewide: bundlerApp makeWrapper buildInputs -> nativeBuildInputs

the docs for bundlerApp used to have makeWrapper in buildInputs but it
has been corrected already

Artturin committed:
cabal2nix: fix cross-eval

for some reason symlinkJoin nativeBuildInputs isn't being spliced correctly

```
nix-repl> cabal2nix
error: attribute 'runtimeShell' missing

       at /nix/store/4hzwal3mxyppnz63rv174iky8j16a7mg-source/pkgs/top-level/all-packages.nix:867:17:

          866|       substitutions = {
          867|         shell = targetPackages.runtimeShell;
             |                 ^
          868|         passthru.tests = tests.makeWrapper;

       … while evaluating the attribute 'shell' of the derivation 'hook'

       at /nix/store/4hzwal3mxyppnz63rv174iky8j16a7mg-source/pkgs/stdenv/generic/make-derivation.nix:270:7:

          269|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          270|       name =
             |       ^
          271|         let

       … while evaluating the attribute 'nativeBuildInputs' of the derivation 'cabal2nix-aarch64-unknown-linux-gnu-2.19.0'

       at /nix/store/4hzwal3mxyppnz63rv174iky8j16a7mg-source/pkgs/stdenv/generic/make-derivation.nix:270:7:

          269|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          270|       name =
             |       ^
          271|         let
«derivation
```

Artturin committed:
cdmemu: fix cross eval

there's still many improvements that can be done to cdemu expressions

Artturin committed:
nodePackages: fix cross-eval of overrides.nix

Artturin committed:
treewide: makeWrapper buildInputs to nativeBuildInputs

most found with https://github.com/siraben/nix-lint

SuperSandro2000 commented on pkgs/development/compilers/opa/default.nix:
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
 
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
```
```suggestion
  buildInputs = [
    perl jdk openssl coreutils zlib ncurses
```

SuperSandro2000 commented on pkgs/development/compilers/opa/default.nix:
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
-    makeWrapper gcc binutils gnumake nodejs
+    gcc binutils gnumake nodejs
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/compilers/opa/default.nix:
```diff
@@ -44,8 +44,9 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
 
+  nativeBuildInputs = [ makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ gcc binutils nodejs makeWrapper which ];
```

SuperSandro2000 commented on pkgs/development/compilers/opendylan/default.nix:
```diff
@@ -13,8 +13,9 @@ stdenv.mkDerivation {
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = (if stdenv.hostPlatform.system == "i686-linux" then [ mps ] else [ boehmgc ]) ++ [
-    opendylan-bootstrap boehmgc gnused autoconf automake perl makeWrapper
+    opendylan-bootstrap boehmgc gnused autoconf automake perl
```
```suggestion
    opendylan-bootstrap boehmgc perl
```

SuperSandro2000 commented on pkgs/development/compilers/opendylan/default.nix:
```diff
@@ -13,8 +13,9 @@ stdenv.mkDerivation {
     fetchSubmodules = true;
   };
 
+  nativeBuildInputs = [ makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ autoconf automake makeWrapper ];
```

SuperSandro2000 commented on pkgs/development/interpreters/lfe/generic-builder.nix:
```diff
@@ -37,7 +37,8 @@ buildRebar3 {
 
-  buildInputs = [ erlang makeWrapper ];
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ erlang ];
```
erlang is a compiler. It probably also goes to nativeBuildInputs.

SuperSandro2000 commented on pkgs/development/interpreters/maude/default.nix:
```diff
@@ -22,9 +22,9 @@ stdenv.mkDerivation {
+  nativeBuildInputs = [ unzip makeWrapper ];
   buildInputs = [
-    flex bison ncurses buddy tecla gmpxx libsigsegv makeWrapper cln yices
+    flex bison ncurses buddy tecla gmpxx libsigsegv cln yices
```
```suggestion
    ncurses buddy tecla gmpxx libsigsegv cln yices
```

SuperSandro2000 commented on pkgs/development/interpreters/maude/default.nix:
```diff
@@ -22,9 +22,9 @@ stdenv.mkDerivation {
   };
 
-  nativeBuildInputs = [ unzip ];
+  nativeBuildInputs = [ unzip makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ bison flex makeWrapper unzip ];
```

SuperSandro2000 commented on pkgs/development/mobile/androidenv/build-tools.nix:
```diff
@@ -2,8 +2,9 @@
-    lib.optionals (os == "linux") [ autoPatchelfHook pkgs.glibc pkgs.zlib pkgs.ncurses5 pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 pkgs.libcxx ];
+  nativeBuildInputs = [ makeWrapper ]
+    ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
+  buildInputs = lib.optionals (os == "linux") [ pkgs.glibc pkgs.zlib pkgs.ncurses5 pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 pkgs.libcxx ];
```
```suggestion
  buildInputs = lib.optionals (os == "linux") [ glibc libcxx ncurses5 zlib pkgs_i686.glibc pkgs_i686.zlib pkgs_i686.ncurses5 ];
```

SuperSandro2000 commented on pkgs/development/mobile/androidenv/tools/25.nix:
```diff
@@ -2,8 +2,8 @@
-  buildInputs = [ autoPatchelfHook makeWrapper ]
-    ++ lib.optional (os == "linux") [ pkgs.glibc pkgs.xorg.libX11 pkgs.xorg.libXext pkgs.xorg.libXdamage pkgs.xorg.libxcb pkgs.xorg.libXfixes pkgs.xorg.libXrender pkgs.fontconfig.lib pkgs.freetype pkgs.libGL pkgs.zlib pkgs.ncurses5 pkgs.libpulseaudio pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig pkgs_i686.freetype pkgs_i686.zlib ];
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
+  buildInputs = lib.optional (os == "linux") [ pkgs.glibc pkgs.xorg.libX11 pkgs.xorg.libXext pkgs.xorg.libXdamage pkgs.xorg.libxcb pkgs.xorg.libXfixes pkgs.xorg.libXrender pkgs.fontconfig.lib pkgs.freetype pkgs.libGL pkgs.zlib pkgs.ncurses5 pkgs.libpulseaudio pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig pkgs_i686.freetype pkgs_i686.zlib ];
```
```suggestion
  buildInputs = lib.optional (os == "linux") [
    glibc xorg.libX11 xorg.libXext xorg.libXdamage xorg.libxcb xorg.libXfixes xorg.libXrender
    fontconfig.lib freetype libGL zlib ncurses5 libpulseaudio 
    pkgs_i686.glibc pkgs_i686.xorg.libX11 pkgs_i686.xorg.libXrender pkgs_i686.fontconfig 
    pkgs_i686.freetype pkgs_i686.zlib
  ];
```

Artturin replied:
Leaving it for the creator @svanderburg

SuperSandro2000 commented on pkgs/development/tools/build-managers/msbuild/default.nix:
```diff
@@ -36,12 +36,12 @@ stdenv.mkDerivation rec {
     dotnet-sdk
     mono
     unzip
+    makeWrapper
   ];
 
   buildInputs = [
     dotnetPackages.Nuget
     glibcLocales
```
```suggestion
    dotnet-sdk
    dotnetPackages.Nuget
    glibcLocales
    makeWrapper
    mono
    unzip
```

SuperSandro2000 commented on pkgs/games/megaglest/default.nix:
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
   buildInputs = [ git curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
```
```suggestion
  buildInputs = [
    curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
```

SuperSandro2000 commented on pkgs/games/megaglest/default.nix:
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ cmake git pkg-config makeWrapper ];
```

SuperSandro2000 commented on pkgs/games/megaglest/default.nix:
```diff
@@ -37,9 +37,9 @@ stdenv.mkDerivation {
+  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
   buildInputs = [ git curl SDL2 xercesc openal lua libpng libjpeg libvlc wxGTK
-    glib cppunit fontconfig freetype ftgl glew libogg libvorbis makeWrapper libGLU ];
+    glib cppunit fontconfig freetype ftgl glew libogg libvorbis libGLU ];
```
```suggestion
    glib cppunit fontconfig freetype ftgl glew libogg libvorbis libGLU
  ];
```

SuperSandro2000 commented on pkgs/games/xconq/default.nix:
```diff
@@ -10,8 +10,9 @@ stdenv.mkDerivation rec {
   };
 
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ cpio xorgproto libX11 libXmu libXaw libXt tcl tk libXext
```
```suggestion
  buildInputs = [
    cpio xorgproto libX11 libXmu libXaw libXt tcl tk libXext fontconfig
  ];
```

SuperSandro2000 commented on pkgs/games/xconq/default.nix:
```diff
@@ -10,8 +10,9 @@ stdenv.mkDerivation rec {
+  nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ cpio xorgproto libX11 libXmu libXaw libXt tcl tk libXext
-    fontconfig makeWrapper ];
+    fontconfig ];
```
```suggestion
```

SuperSandro2000 commented on pkgs/tools/filesystems/glusterfs/default.nix:
```diff
@@ -16,7 +16,7 @@ let
   buildInputs = [
     fuse bison flex openssl ncurses readline
```
```suggestion
    fuse openssl ncurses readline
```

SuperSandro2000 commented on pkgs/tools/filesystems/glusterfs/default.nix:
```diff
@@ -16,7 +16,7 @@ let
   buildInputs = [
     fuse bison flex openssl ncurses readline
     libtool pkg-config zlib libaio libxml2
```
```suggestion
    zlib libaio libxml2
```

SuperSandro2000 commented on pkgs/tools/filesystems/glusterfs/default.nix:
```diff
@@ -106,7 +106,7 @@ in stdenv.mkDerivation rec {
   ];
 
-  nativeBuildInputs = [ rpcsvc-proto autoconf automake ];
+  nativeBuildInputs = [ rpcsvc-proto autoconf automake makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ autoconf automake bison flex libtool makeWrapper pkg-config rpcsvc-proto ];
```

SuperSandro2000 commented on pkgs/tools/security/wpscan/default.nix:
```diff
@@ -9,9 +9,7 @@ bundlerApp {
-  buildInputs = [
-    makeWrapper
-  ];
+  nativeBuildInputs = [ makeWrapper ];
```
Please revert

Artturin replied:
added support for nativeBuildInputs in [`3bfda6b` (#186564)](https://github.com/NixOS/nixpkgs/pull/186564/commits/3bfda6b18db43278f240d6598ba048c29375ec98)

Artturin commented on pkgs/applications/emulators/cdemu/base.nix:
```diff
@@ -1,12 +1,12 @@
     sha256 = pkgSha256;
   };
-  nativeBuildInputs = [ pkg-config cmake ];
+  nativeBuildInputs = (attrs.nativeBuildInputs or [ ]) ++ [ pkg-config cmake ];
```
nope
```
error: attribute 'nativeBuildInputs' missing

       at /nix/store/mzqhdx1sk6qxia1qvw7chvi8bbsjp9py-source/pkgs/applications/emulators/cdemu/base.nix:9:23:

            8|   };
            9|   nativeBuildInputs = attrs.nativeBuildInputs ++ [ pkg-config cmake ];
             |                       ^
           10|   setSourceRoot = ''
«derivation
```

if the `nativeBuildInputs ? []` is removed then 

```
nix-repl> cdemu-client
error: anonymous function at /nix/store/9czmzcsnb3lxk6dk1bv9j238s8yylzfl-source/pkgs/applications/emulators/cdemu/base.nix:2:1 called with unexpected argument 'nativeBuildInputs'

       at /nix/store/9czmzcsnb3lxk6dk1bv9j238s8yylzfl-source/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|
```

SuperSandro2000 replied:
Not sure if that is nicer or not

Artturin replied:
@bendlas

Artturin commented on pkgs/applications/gis/grass/default.nix:
```diff
@@ -15,9 +15,9 @@ stdenv.mkDerivation rec {
+  readline ffmpeg netcdf geos postgresql libmysqlclient blas
   libLAS proj-datumgrid zstd wrapGAppsHook ]
     ++ lib.optionals stdenv.isLinux [ cairo pdal wxGTK31 ]
     ++ lib.optional stdenv.isDarwin wxmac
```
the fails to apply lol

Leaving it for the maintainers @mpickering @willcohen

Artturin committed:
bundlerApp: accept nativeBuildInputs

https://github.com/nixos/nixpkgs/commit/69dcb1a2c0c5596fbbbe7fa50d4f8277c6f1ea6a

added support for buildInputs with the intention of adding makeWrapper
but makeWrapper should be in nativeBuildInputs

Artturin commented on pkgs/applications/science/math/pspp/default.nix:
```diff
@@ -13,10 +13,10 @@ stdenv.mkDerivation rec {
 
-  nativeBuildInputs = [ pkg-config texinfo python3 ];
+  nativeBuildInputs = [ pkg-config texinfo python3 makeWrapper ];
   buildInputs = [ libxml2 readline zlib perl cairo gtk3 gsl
```
no formatting pls, hopefully we will get a treewide format sometime..

ck3d reviewed (approved):
LGTM
